networkx/linalg/attrmatrix.py

```diff
@@ -141,6 +142,7 @@ def value(u, v):
     return value
 
 
+@nx._dispatch(edge_attrs="edge_attr", node_attrs="node_attr")
 def attr_matrix(
```

Comment:
How can we indicate an attribute _must_ exists for all edges or nodes? In `attr_matrix` and `attr_sparse_matrix`, `weight=None` (the default) means each edge has a weight of 1 (or count for multigraph), otherwise the edge attribute must be present for all edges.